src/verilog.y

```diff
@@ -4447,6 +4447,11 @@ idArrayed<nodep>:		// IEEE: id + select
 		id						{ $$ = new AstParseRef($<fl>1,VParseRefExp::PX_TEXT,*$1,NULL,NULL); }
 	//			// IEEE: id + part_select_range/constant_part_select_range
 	|	idArrayed '[' expr ']'				{ $$ = new AstSelBit($2,$1,$3); }  // Or AstArraySel, don't know yet.
+	|       idArrayed '[' yP_PLUSPLUS   expr ']'		{ $$ = new AstSelBit($2,$1,new AstPreAdd($3,$4, new AstConst($3, 1))); }
+	|       idArrayed '[' yP_MINUSMINUS expr ']'		{ $$ = new AstSelBit($2,$1,new AstPreAdd($3,$4, new AstConst($3, -1))); }
+	|       idArrayed '[' expr yP_PLUSPLUS ']'		{ $$ = new AstSelBit($2,$1,new AstPostAdd($4,$3, new AstConst($4, 1))); }
+	|       idArrayed '[' expr yP_MINUSMINUS ']'		{ $$ = new AstSelBit($2,$1,new AstPostAdd($4,$3, new AstConst($4, -1))); }
```

Comment:
Wetry to match the IEEE grammar layout where possible. Can you look at moving these under inc_or_dec_expression (which was commented out and has the right rules but needs these new productions to make AstPre/PostAdd), then comment in the rule inc_or_dec_expression under expr and anywhere else.

This will mean you can get AstPre/PostAdd in pretty much anywhere.  Then in the visitors, give an error if it is something unsupported rather than relying on the grammar.

Reply:
I don't think it will be that simple. I tried uncommenting the nodes under `inc_or_dec_expression` filling them with the new AstNodes and then doing a simple `{ $$ = $1}` under `expr<nodep>`, but his gives lots of conflicts. We will probably need more time to debug this.